tools/data-handler/src/calculate.ts

```diff
@@ -398,43 +323,41 @@ fieldtype(X, Field, "cardkeys") :- field(X, Field, _), card(Value) : field(X, Fi
   /**
    * Runs a logic program.
    *
-   * @param {string} projectPath Path to a project
-   * @param {string} cardKey Optional, if missing the calculations are run for the whole cardtree.
-   *                         If defined, calculates only subtree.
+   * @param projectPath Path to a project
+   * @param filePath Path to a query file to be run in relation to current working directory
    * @returns parsed program output
    */
   public async run(
     projectPath: string,
-    cardKey: string,
-  ): Promise<ParseResult[] | undefined> {
+    filePath: string,
+  ): Promise<ParseResult | undefined> {
     Calculate.project = new Project(projectPath);
-
-    const card = await Calculate.project.findSpecificCard(cardKey);
-    if (!card) {
-      throw new Error(`Card '${cardKey}' not found`);
-    }
-
-    const text = `
-            #show.
-            #show field(Cardkey, Field, Value):
-                field(Cardkey, Field, Value),
-                Cardkey = ${card.key},
-                not userfield(Cardkey, Field).
-            #show fieldtype(Cardkey, Field, Fieldtype):
-                fieldtype(Cardkey, Field, Fieldtype),
-                Cardkey = ${card.key},
-                not userfield(Cardkey, Field).`;
     const main = join(
       Calculate.project.calculationFolder,
       Calculate.mainLogicFileName,
     );
-    const clingo = spawnSync(
-      this.logicBinaryName,
-      ['-', '--outf=0', '--out-ifs=\\n', '-V0', `${main}`],
-      { encoding: 'utf8', input: text },
+    const queryLanguage = join(
+      Calculate.project.calculationFolder,
+      Calculate.queryLanguageFileName,
     );
 
+    const args = [
+      '-',
+      '--outf=0',
+      '--out-ifs=\\n',
+      '-V0',
+      main,
+      queryLanguage,
+      filePath,
+    ];
+    const clingo = spawnSync(this.logicBinaryName, args, {
+      encoding: 'utf8',
+    });
+    // print the command
+    console.log(`Ran command: ${this.logicBinaryName} ${args.join(' ')}`);
+
     if (clingo.stdout) {
+      console.log(`Clingo output: \n${clingo.stdout}`);
```

Comment:
Do we need to state that it is "clingy" output? Would "Calculation output" do?

Reply:
Console logging the outputs should be a temporary solution anyway. Maybe it's a good reminder to fix it sooner or later? And as a debug log, I don't think it's a bad idea to specify it is clingo output, because then the one who debugs knows it is clingo and not clingraph for example.